### ISSUE TITLE
Add-on not working on Firefox

### ISSUE THREAD

**Issue by kmf** [jplattel/obsidian-clipper#69]
![image](https://user-images.githubusercontent.com/78700/212958634-624dfd79-27a1-4cfe-b069-8856e7ffd9da.png)

Tested on Firefox 109.0

Old Version (0.3.6): https://addons.mozilla.org/en-US/firefox/addon/obsidian-clipper/

**Comment by ondrahlavac** [comment]
Not sure about @jplattel's plans, but to me it feels like chrome/firefox compatibility is not comming back anytime soon.
I mean, I'm an amateur but the firefox build is full of chrome specific code. 😄 

I see two options.
 - Fork the firefox version into it's own repo and rewrite it.
 - Or maybe discuss some webextension polyfills with @jplattel  and update the code to work on both platforms
 
I'm trying to hack my version locally at the monent.
Will report on my results. 😆

**Comment by ondrahlavac** [comment]
Ok, so I'm back with the news.

There has been an issue with new version of manifest. Google pushed for the quick adoption of Manifest V3 and Firefox has not finalized their implementation.

Nevertheless, two things need to change for this to work with Firefox.
 1. _manifest.json_ for the firefox build needs to keep the following definiton, as it was in Manifest V2:
`"background": {
    "scripts": ["background.js"],
    "type": "module"
},`
 2. Since the new **chrome.storage.sync** is being used, firefox needs AppID to be added to _manifest.json_.
 `    "browser_specific_settings": {
        "gecko": {
          "id": "firefox-obsidian-clipper-addon@example.com"
        }
    }`
    
Normally, I would send a pull request, but this is something for Joost (@jplattel) to decide.
Basically you need to build with two different manifests for firefox and chrome.

I got my version running as unsigned addon - https://github.com/ondrahlavac/obsidian-clipper/tree/personal-version
But it's just a hack, not a fix.


So... @jplattel, can I somehow be of service so we can get this fixed? :-)

**Comment by ondrahlavac** [comment]
@kmf @mcheck
You can use my hacked version - [obsidian_clipper_for_firefox-0.5.x.zip](https://github.com/ondrahlavac/obsidian-clipper/blob/personal-version/build/firefox/obsidian_clipper_for_firefox-0.5.x.zip)
https://github.com/ondrahlavac/obsidian-clipper/tree/personal-version/

Just to carry us over before we find a way to get it all sorted and get a fresh new version to addons.mozilla.org
